### PR TITLE
feat: add support for secrets

### PR DESCRIPTION
**Pull request by ascorbic**
WIP: adding helpers to allow secrets to be injected, based currently on env vars. Can be installed as `@netlify/functions@handle-secrets`

### CODE REVIEW

File: src/lib/services.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
+  "github": {
+    "tokens": {
+      "token": "ONEGRAPH_GITHUB_TOKEN"
```
Comment (ascorbic) [inline src/lib/services.json:4]: Here `"token"` is the name of the property exposed on the `secrets` object. There could in theory be multiple tokens per service: pass them as keys in the `"tokens"` object.

**Review by lindsaylevine — COMMENTED** [review]
NEAT

File: package.json [hunk 2]
```diff
@@ -2,11 +2,12 @@
   "name": "@netlify/functions",
   "main": "./src/main.js",
   "types": "./src/main.d.ts",
-  "version": "0.7.2",
+  "version": "0.7.3-handle-secrets.1",
```
Comment (tiffanosaurus) [inline package.json:5]:
```suggestion
  "version": "0.7.3-handle-secrets.2",
```
```suggestion
  "version": "0.7.3-handle-secrets.1",
```

File: README.md [hunk 3]
```diff
@@ -74,6 +74,108 @@ The following types are exported:
 - `HandlerEvent`
 - `HandlerResponse`
 
+
+## Auth Management
```
Comment (tiffanosaurus) [inline README.md:78]:
```suggestion
## Auth Token Manager
```

File: README.md [hunk 4]
```diff
@@ -74,6 +74,108 @@ The following types are exported:
...
+        body: JSON.stringify(result),
+        headers: {
+          "Content-Type": "application/json",
+        },
+      };
+    };
+  ```
+
+### Checking additional metadata about auth token in your functions and site builds
+Auth Management also tracks metadata for installed auth tokens. You can verify that an auth has been installed with the correct scopes before calling into an API (say, for example, to give a better error message in the developer logs). Here's an example:
```
Comment (tiffanosaurus) [inline README.md:132]:
```suggestion
Auth Token Manager also tracks metadata for installed auth tokens. You can verify that an auth has been installed with the correct scopes before calling into an API (say, for example, to give a better error message in the developer logs). Here's an example:
```

File: README.md [hunk 5]
```diff
@@ -74,6 +74,108 @@ The following types are exported:
...
+    }
+  ```
+
+### Accessing integration auth tokens during development
+
+When running your site under `netlify dev`, the environmental variables that power the auth management will be synced, and you can transparently develop against the third party APIs - no additional configuration necessary!
+
+### Updating or removing auth tokens from your site
+
+At any time you can revisit the Auth Management tab for your site in [Netlify Labs](https://app.netlify.com/user/labs) (select your profile avatar, then Netlify Labs) to see the installed auth. From there, you can select new scopes for already-installed auth and then run through the browser-based auth flow again, and the new scopes will be available to all your existing, deployed functions and site builds _instantly_.
```
Comment (tiffanosaurus) [inline README.md:175]:
```suggestion
At any time you can revisit the Auth Token Manager tab for your site in [Netlify Labs](https://app.netlify.com/user/labs) (select your profile avatar, then Netlify Labs) to see the installed auth. From there, you can select new scopes for already-installed auth and then run through the browser-based auth flow again, and the new scopes will be available to all your existing, deployed functions and site builds _instantly_.
```

**Pull request closed without merging**